### ISSUE TITLE
add ramp gradient support for particles

### ISSUE THREAD

**Issue by deltakosh** [BabylonJS/Babylon.js#5064]
add two new functions to CPU particles:
- addRampGradient(index, color): this will define a curve of colors between 0 and 1
- addAlphaSmoothStep(index, min, max): this will define a curve for alpha values

The goal is (when this will be turned on with a property named `system.enableRampGradients`) to compute the final output color using this formula:

`outputColor = particleColor * textureColor * rampGradientColor(AlphaSmoothStep(textureAlpha))`

In other words, the texture alpha will be used as an index on the alphaSmoothStep curve transforming it from [0,1] range to [min, max] range. The transformed alpha will then be used as an index in the ramp gradient curve.
